src/context/viewer-context.tsx

```diff
@@ -179,6 +183,18 @@ function useAuthedViewer() {
     setViewer(authenticatedUser)
   }
 
+  const [ability, setAbility] = React.useState(canDoNothingAbility)
+  React.useEffect(() => {
+    const fetchAbility = async (authToken: string) => {
+      const ability = await getAbilityFromToken(authToken)
+      setAbility(ability)
+    }
+
+    if (authToken) {
+      fetchAbility(authToken)
+    }
+  }, [authToken])
+
```

Comment:
This uses `src/server/ability.ts` to build the `ability` object. That object is then included on the `viewer` context.